docs/source/reducing_memory_usage.md

```diff
@@ -268,6 +268,168 @@ training_args = RLOOConfig(..., ds3_gather_for_generation=False)
 
 This adjustment prevents model weights from being gathered, avoiding OOM errors, but it may result in slower generation speeds.
 
+## Context Parallelism
+
+Context Parallelism (CP) is a parallelization technique that enables training with longer sequences by splitting the sequence dimension across multiple GPUs. Each GPU processes a portion of the sequence, allowing you to train with sequences longer than what would fit on a single GPU's memory.
+
+For more details on Context Parallelism, see the [Ultrascale Playbook - Context Parallelism](https://huggingface.co/spaces/nanotron/ultrascale-playbook?section=context_parallelism).
+
+Context parallelism is particularly useful when:
+- You want to train with very long sequences (>2048 tokens)
+- Single GPU memory is insufficient for your desired sequence length
+- You need to maintain sequence coherence across the full context
+
+### Requirements and Limitations
+
+Context Parallelism has specific requirements:
+
+1. **FSDP2 (PyTorch FSDP v2)** is required as the distributed training backend
+2. **SDPA attention** - Flash Attention is currently not supported with Context Parallelism
+3. **Sequence length divisibility** - sequences must be divisible by `cp_size * 2`
+
+### Configuration
+
+To enable Context Parallelism, you need to configure both Accelerate and your training arguments:
+
+#### Accelerate Configuration
+
+Create an accelerate config file (e.g. `context_parallel_config.yaml` for 2 GPUs):
```

Comment:
I wonder if it makes sense to have a copy of this in `accelerate_configs/fsdp2_cp.yaml` so there's a standard reference people can work from? If you agree, I'd make it to run on 8 GPUs which is the default for our other configs